### PR TITLE
permission description edit

### PR DESCRIPTION
**Pull request by xnetsrak**
## Description

permission description edit
## References

* #0000

### CODE REVIEW

File: source/Energinet.DataHub.MarketParticipant.Application/Validation/UpdatePermissionCommandRuleSet.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+namespace Energinet.DataHub.MarketParticipant.Application.Validation;
+
+public sealed class UpdatePermissionCommandRuleSet : AbstractValidator<UpdatePermissionCommand>
+{
+    public UpdatePermissionCommandRuleSet()
+    {
+        RuleFor(command => command)
+            .NotNull();
+
+        RuleFor(command => command.Id)
```
Comment (defectiveAi) [inline source/Energinet.DataHub.MarketParticipant.Application/Validation/UpdatePermissionCommandRuleSet.cs:27]: Check if id is part of enum.

File: source/Energinet.DataHub.MarketParticipant.Application/Validation/UpdatePermissionCommandRuleSet.cs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+            .NotEmpty();
+
+        RuleFor(command => command.Description)
```
Comment (defectiveAi) [inline source/Energinet.DataHub.MarketParticipant.Application/Validation/UpdatePermissionCommandRuleSet.cs:30]: Check for max. length limit of description.

File: source/Energinet.DataHub.MarketParticipant.Domain/Repositories/IPermissionRepository.cs [hunk 3]
```diff
@@ -35,4 +36,11 @@ public interface IPermissionRepository
     /// <param name="eicFunction">The eicFunction you want to get permissions for.</param>
     /// <returns>List of permissions for this EicFunction.</returns>
     Task<IEnumerable<PermissionDetails>> GetToMarketRoleAsync(EicFunction eicFunction);
+
+    /// <summary>
+    /// Update permission
+    /// </summary>
+    /// <param name="permissionToUpdate">permission to update</param>
+    /// <param name="newDescription">New permission description</param>
+    Task UpdatePermissionAsync(Permission permissionToUpdate, string newDescription);
```
Comment (defectiveAi) [inline source/Energinet.DataHub.MarketParticipant.Domain/Repositories/IPermissionRepository.cs:45]: https://learn.microsoft.com/en-us/dotnet/architecture/microservices/microservice-ddd-cqrs-patterns/infrastructure-persistence-layer-design

This method is not correct repository pattern. Proper pattern would be to take a single aggregate (in this case PermissionDetails, no the best of names) as input.

So `Task UpdateAsync(PermissionDetails permissionDetails)`

Comment (xnetsrak) [reply source/Energinet.DataHub.MarketParticipant.Domain/Repositories/IPermissionRepository.cs:45]: Im writing my own book

**Review by defectiveAi — APPROVED** [review]

**Pull request merged**